dali/base/dadfs.cpp

```diff
@@ -3472,6 +3485,7 @@ protected: friend class CDistributedFilePart;
             queryAttributes().setPropInt64("@size", totalsize);
         if (useableCheckSum)
             queryAttributes().setPropInt64("@checkSum", checkSum);
+        setHistory(fdesc);
```

Comment:
Either it should always create (and you do) and return a reference so interface is consistent.
Or it should not create blank History nodes, which are not used the vast majority of the time and end up being serialized and committed to meta store.

It was I thought going in the direction of not insisting it was there to avoid waste, therefore returning it optionally made sense.
If you are going to force it's creation, and then the only case it's not there is legacy, it doesn't make sense to return a pointer which will be null only for legacy. It should be consistent and if create most of time, create all the time.

It's not a big deal to have the extra empty usually unused History node serialize/part of meta store, so leave it like it is, but make sure always exists (will only not in legacy), add a comment to say why creating in that case (in queryHistory()).